sympy/matrices/matrices.py

```diff
@@ -1581,6 +1583,12 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
+        num = sympify(num)
+        if not num.is_integer and (num >= 0) == True and not num % 1 == 0:
```

Comment:
I wonder what should be done if `num < 0`. Also, what is the purpose of `not num % 1 == 0` if `not num.is_integer` is true?